Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Apple.cpp

```diff
@@ -89,15 +89,15 @@ bool pfApplePasswordStore::SetPassword(const ST::string& username, const ST::str
     CFStringRef serviceName = CFStringCreateWithSTString(service);
     CFDataRef   passwordData = CFDataCreate(nullptr, (const UInt8*)password.c_str(), password.size());
 
-    CFAutorelease(accountName);
-    CFAutorelease(serviceName);
-    CFAutorelease(passwordData);
+    hsAutorelease(accountName);
+    hsAutorelease(serviceName);
+    hsAutorelease(passwordData);
```

Comment:
My guess is that Autorelease was just used here because it's convenient. We might need to change some of the ordering but we could use 'CFRelease' here to be more portable?

(Same with the other uses.)

Reply:
I don't remember if these were originally `CFRelease` and got changed to `CFAutorelease` when you refactored the security APIs (which don't compile on Leopard btw). There's no explicit autorelease pool here so I don't know what benefit `CFAutorelease` is giving us.

I'm fine with doing the releasing manually here since all the object lifecycles should be contained to within this function.

Reply:
I'd like to trim it down because I want to see what's really leaning on autorelease. The abstraction could be avoided if nothing needs it. I can do a more full pass on that later.

Generally - compiling something without ARC and putting back in the backwards compatibility memory management would be preferred to creating a bunch of new abstraction IMO. I use ARC in the client because we had some weird memory management going on and it was easier just to turn on ARC. But we don't need to maintain ARC support in every file.

Reply:
(Along with that - I should warn there is always the possibility we'll need to adopt Swift for parts of the client in the future. So if we're thinking about backwards compatibility we may also want to decide what is core to Plasma and what could diverge for older macOS support.)

Reply:
It sounds like there's general agreement to move the pfPasswordStore stuff to not using autorelease at all, which makes sense to me. I'm happy to make that change if you don't see any concerns with how the code is currently structured

Reply:
Let's go ahead and make that change. I'd prefer to not define custom scaffolding where possible - and it doesn't hurt us in any way on present macOS versions.